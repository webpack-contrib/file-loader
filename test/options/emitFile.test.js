/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('emitFile', () => {
    describe('{Boolean}', () => {
      test('True (Default)', async () => {
        const config = {
          loader: {
            test: /(png|jpg|svg)/,
            options: {
              emitFile: true,
            },
          },
        };

        const stats = await webpack('emitFile/fixture.js', config);
        const { assets } = stats.toJson().modules[1];

        expect(assets[0]).toMatchSnapshot();
      });

      test('False', async () => {
        const config = {
          loader: {
            test: /(png|jpg|svg)/,
            options: {
              emitFile: false,
            },
          },
        };

        const stats = await webpack('emitFile/fixture.js', config);
        const { assets } = stats.toJson().modules[1];

        expect(assets[0]).toBe(undefined);
      });
    });
  });
});
