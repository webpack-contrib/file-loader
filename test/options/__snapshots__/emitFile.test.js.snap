// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options emitFile {Boolean} True (Default) 1`] = `"64eababb117f90535085779cc0325003.svg"`;
